pgspecial/dbcommands.py

```diff
@@ -28,36 +30,44 @@
 
 @special_command("\\l", "\\l[+] [pattern]", "List databases.", aliases=("\\list",))
 def list_databases(cur, pattern, verbose):
-    query = '''SELECT d.datname as "Name",
+    params = {}
+    query = SQL(
+        """SELECT d.datname as "Name",
         pg_catalog.pg_get_userbyid(d.datdba) as "Owner",
         pg_catalog.pg_encoding_to_char(d.encoding) as "Encoding",
         d.datcollate as "Collate",
         d.datctype as "Ctype",
-        pg_catalog.array_to_string(d.datacl, E'\n') AS "Access privileges"'''
+        pg_catalog.array_to_string(d.datacl, E'\n') AS "Access privileges"
+        {verbose_fields}
+        FROM pg_catalog.pg_database d
+        {verbose_tables}
+        {pattern_where}
+        ORDER BY 1"""
+    )
     if verbose:
-        query += ''',
+        params["verbose_fields"] = SQL(
+            ''',
             CASE WHEN pg_catalog.has_database_privilege(d.datname, 'CONNECT')
                     THEN pg_catalog.pg_size_pretty(pg_catalog.pg_database_size(d.datname))
                     ELSE 'No Access'
             END as "Size",
             t.spcname as "Tablespace",
             pg_catalog.shobj_description(d.oid, 'pg_database') as "Description"'''
-    query += """
-    FROM pg_catalog.pg_database d
-    """
-    if verbose:
-        query += """
-        JOIN pg_catalog.pg_tablespace t on d.dattablespace = t.oid
-        """
-    params = {}
+        )
+        params["verbose_tables"] = SQL(
+            """JOIN pg_catalog.pg_tablespace t on d.dattablespace = t.oid"""
+        )
+    else:
+        params["verbose_fields"] = SQL("")
+        params["verbose_tables"] = SQL("")
+
     if pattern:
-        query += """
-        WHERE d.datname ~ %(schema_pattern)s
-        """
         _, schema = sql_name_pattern(pattern)
-        params["schema_pattern"] = schema
-    query = query + " ORDER BY 1"
-    cur.execute(query, params)
+        params["pattern_where"] = SQL("""WHERE d.datname ~ {}""").format(schema)
+    else:
+        params["pattern_where"] = SQL("")
+    log.debug(query.format(**params).as_string(cur))
```

Comment:
Calling `.format` twice, once to log, and once to execute, could save that formatted instance and reuse.